src/lib.rs

```diff
@@ -196,7 +205,10 @@ fn replace_dir<P: AsRef<Path>>(old: P, new: P) -> Result<()> {
         old_as_file.sync_all()
             .chain_err(|| format!("failed to sync old dir '{}'", old.display()))?;
 
-        truncate_dir(new)?;
+        if new.exists() {
+            fs::rename(new, stage)
+                .chain_err(|| format!("failed to move '{:?}' to '{:?}'", new, stage))?;
```

Comment:
If this code fails once and leaves the stage dir in place, all subsequent invocations will fail until the stage dir is cleaned out by hand.  Is that worth fixing?

Reply:
I think adding a call to `truncate_dir(stage)` before trying to do the renaming should be sufficient so we might as well fix it.